src/main/kotlin/com/geospock/pubvote/GroupSplitter.kt

```diff
@@ -9,13 +11,25 @@ import java.util.Collections
 class GroupSplitter(private val maxGroupSize: Int) {
 
     fun splitGroups(attending: Group): List<Group> {
-        val groups = (attending.size + maxGroupSize - 1) / maxGroupSize
-
-        val listToSplit = attending.toMutableList()
-        Collections.shuffle(listToSplit)
-        return listToSplit.withIndex()
-                .groupBy {
-                    it.index % groups
-                }.map { it.value.map { it.value } }
+        val groups = (1 .. (attending.size + maxGroupSize - 1) / maxGroupSize)
+                .map { mutableListOf<People>() }
+
+        groups.addFromRole(attending, Role.ELT)
+        groups.addFromRole(attending, Role.CSMT)
+        groups.addFromRole(attending, Role.ESMT)
+        groups.addFromRole(attending, Role.PEON)
+
+        return groups
+    }
+
+    private fun List<MutableList<People>>.addFromRole(attending: Group, role: Role) {
+        val peopleToAdd = attending.filter { person -> person.role == role }
+        Collections.shuffle(peopleToAdd)
+
+        peopleToAdd.forEach {
+            this.minBy { it.size }!!
```

Comment:
why not `this.minBy { it.size }?.add(it)`?

Reply:
👍

Reply:
fancy seeing you here - that is safer - but if for some reason minBy doesn't return a group I'd rather if failed early, but maybe I can throw my own exception instead of an NPE?

Reply:
It can only happen if the receiver is an empty list, so `IllegalArgumentException` would be ok I think.